tests/test_volume.py

```diff
@@ -358,6 +417,39 @@ def test_rotate(L, dtype):
         assert np.allclose(ref_vol, rot_vol, atol=utest_tolerance(dtype))
 
 
+def test_rotate(vols_hot_cold):
+    """
+    We rotate Volumes containing random hot/cold spots by random rotations and check that
+    hot/cold spots in the rotated Volumes are in the expected locations.
+    """
+    vols, hot_cold_locs, vol_center = vols_hot_cold
+    dtype = vols.dtype
+    L = vols.resolution
+
+    # Generate random rotations.
+    rots = Rotation.generate_random_rotations(n=vols.n_vols, dtype=dtype)
+    rots_mat = rots.matrices
+
+    # Expected location of hot/cold spots relative to (0, 0, 0) origin in (x, y, z) order.
+    expected_hot_cold = np.transpose(
+        rots_mat @ np.transpose(hot_cold_locs[..., ::-1], axes=(0, 2, 1)),
+        axes=(0, 2, 1),
+    )
+
+    # Expected location of hot/cold spots relative to Volume center (L/2, L/2, L/2) in (z, y, x) order.
+    expected_locs = np.round(expected_hot_cold[..., ::-1] + vol_center)
+
+    # Rotate Volumes.
+    rotated_vols = vols.rotate(rots)
+
+    # Check that new hot/cold spots are in expectecd locations.
+    for i in range(vols.n_vols):
+        new_hot_loc = np.unravel_index(np.argmax(rotated_vols.asnumpy()[i]), (L, L, L))
```

Comment:
just so lines are easier to follow, `v = rotated_vols.asnumpy()[i]`

Reply:
Thanks, much easier to follow. Done.